frontend/src/components/dialogs/CreateExpenseDialog.jsx

```diff
@@ -34,7 +35,8 @@ export default function CreateExpenseDialog(props) {
 
   function addExpense(expense) {
     console.log("[CrateExpenseDialog]: Will add expense: ", expense);
-    setExpenses([...expenses, expense]);
+    let expenses = expensesState.getState();
+    expensesState.setState([...expenses, expense]);
```

Comment:
Doesn't make much sense.  
```suggestion
    expensesState.modifyExpense(expense]); // for add/update/remove
    // and in the ExpensesState, the business logic can handle the udpate of the state internally with setState
```